arch/posix/include/posix_soc_if.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2017 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef _POSIX_CORE_SOC_PROVIDED_IF_H
+#define _POSIX_CORE_SOC_PROVIDED_IF_H
+
+/*
+ * This file lists the functions the POSIX architecture core expects the
+ * SOC or board will provide
+ *
+ * All functions listed here must be provided by the implementation of the SOC
+ * or all its boards
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void ps_print_error_and_exit(const char *format, ...);
+void ps_print_warning(const char *format, ...);
+void ps_print_trace(const char *format, ...);
+
+void ps_halt_cpu(void);
+void ps_atomic_halt_cpu(unsigned int imask);
+
+
+#include "soc_irq.h" /*Must exist and define _ARCH_IRQ/ISR_* macros*/
+
+unsigned int _arch_irq_lock(void);
+void _arch_irq_unlock(unsigned int key);
+void _arch_irq_enable(unsigned int irq);
+void _arch_irq_disable(unsigned int irq);
+int  _arch_irq_is_enabled(unsigned int irq);
+unsigned int ps_irq_lock(void);
```

Comment:
Do we really need this "`ps_`" prefix here? If we do then I would be in favour of using "`posix_`" or "`native_`" if it's common to all native builds.

Reply:
pc = posix core
ps = posix soc
pb = posix board
The prefix is just for "namespacing", and indicating which layer/module implements the function.
Using 2 letters was just to keep it short.

Reply:
I personally dislike those prefixes. Could we simply use `posix_` everywhere, regardless of the core/soc/board distinction?

Reply:
pc sounds like 'personal computer' :) Agree with @carlescufi here, posix makes more sense now that I think about it.